awx/main/tasks.py

```diff
@@ -1196,7 +1191,12 @@ def run(self, pk, **kwargs):
                 )
                 copy_tree(cwd, os.path.join(private_data_dir, 'project'))
                 ansible_runner.utils.dump_artifacts(params)
-                manager_instance = isolated_manager.IsolatedManager(env, **_kw)
+                manager_instance = isolated_manager.IsolatedManager(
+                    env,
+                    cancelled_callback=lambda: self.update_model(self.instance.pk).cancel_flag,
```

Comment:
We weren't passing `cancelled_callback` to the underlying isolated manager code, so it wasn't checking.

The rest of this PR is just cleanup.

After this merges and is verified, I'm going to actually totally get rid of `awx.main.expect.run` and replace all of our usage w/ runner so that the isolated manager uses *runner* programmatically to run its `_isolated.yml` playbooks (instead of `pexpect`).